wagtail/wagtailcore/models.py

```diff
@@ -567,11 +562,48 @@ def get_context(self, request, *args, **kwargs):
             'request': request,
         }
 
+    @property
+    def template(self):
+        warnings.warn(
+            "The automatically generated template attribute is deprecated - "
+            "use Page.get_template() or set an explicit template attribute .",
+            RemovedInWagtail14Warning)
+        return "%s/%s.html" % (self._meta.app_label, camelcase_to_underscore(self._meta.object_name))
+
+    @property
+    def ajax_template(self):
+        warnings.warn(
+            "The automatically generated ajax_template attribute is deprecated - "
+            "use Page.get_template() or set an explicit ajax_template attribute .",
+            RemovedInWagtail14Warning)
+        return "%s/%s.html" % (self._meta.app_label, camelcase_to_underscore(self._meta.object_name))
+
     def get_template(self, request, *args, **kwargs):
-        if request.is_ajax():
-            return self.ajax_template or self.template
-        else:
+        """
+        Get the template name for this Page class.
+
+        * If `request.is_ajax()`, and the Page has an `ajax_template`
+          attribute, that template will be used.
+        * Otherwise, if the Page has a `template` attribute, that will be used.
+        * Otherwise, a template name of `"app/model_name.html"` will be
+          generated.
+        """
+        # RemovedInWagtail14Warning: `is_auto` and `is_ajax_auto`.
+        # `template` and `ajax_template` attributes will just be set to None
+        # in Wagtail 1.3, and the `is_auto` and `is_ajax_auto` variables
```

Comment:
Oops, missed that one when I was making the RemovedInWagtail14Warning. Fixed now